synapse/handlers/federation.py

```diff
@@ -562,6 +562,10 @@ async def do_invite_join(
             joinee: The User ID of the joining user.
 
             content: The event content to use for the join event.
+
+        Raises:
+            PartialStateConflictError if homeserver was already in the room and it is
+                no longer partial stated.
```

Comment:
I can see this is taken from the comment below, but it hard to understand when this occurs. Is the point that

1. HS is partially joined to the room
2. This function gets called
3. We wait for a response from the remote HS
4. The resync for this room completes; the room is now no longer partially stated
5. We get a response from (3); but we should discard it and create a local join ourselves.

Reply:
That's right (or was right prior to #14882, which guarantees no `PartialStateConflictError`s).